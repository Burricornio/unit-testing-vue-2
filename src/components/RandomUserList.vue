<template>
  <div>
    <div v-for="(person, key) in people" :key="key">
      <div class="person">
        <div class="person__header">
          <img
            v-bind:src="person.picture.large"
            v-bind:alt="person"
            class="rounded img-thumbnail"
          />
          <div class="person__name">
            {{ person.name.first }} {{ person.name.last }}
          </div>
        </div>
        <div class="person__email">
          <a v-bind:href="'mailto:' + person.email">{{ person.email }}</a>
        </div>
        <div class="person__address">
          <address>
            {{ person.location.street.number }}
            {{ person.location.city }}
            {{ person.location.state }}
          </address>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import axios from 'axios'

@Component
export default class RandomUserList extends Vue {
  people = []

  async created(): void {
    try {
      const response = await axios.get(`https://randomuser.me/api/?results=6`)
      this.people = response.data.results
      console.log(response.data)
      // eslint-disable-next-line no-empty
    } catch (error) {}
  }
}
</script>

<style scoped></style>
