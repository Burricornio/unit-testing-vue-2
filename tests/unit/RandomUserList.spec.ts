import { shallowMount, Wrapper } from '@vue/test-utils'
import RandomUserList from '@/components/RandomUserList.vue'
import flushPromises from 'flush-promises'

jest.mock('axios', () => ({
  get: () =>
    Promise.resolve({
      data: {
        results: [
          {
            picture: {
              large: '',
            },
            name: {
              first: '',
            },
            location: {
              street: {
                number: '',
              },
            },
          },
        ],
      },
    }),
}))

describe('RandomUserList suite', () => {
  let wrapper: Wrapper<RandomUserList>

  beforeEach(() => {
    wrapper = shallowMount(RandomUserList)
  })
  // Nos aseguramos de quew nadie ha modificado el componente sin modificsar los tests
  it('Should match snapshots', () => {
    expect(wrapper).toMatchSnapshot()
  })

  // Hacemos un mock de la peticiÃ³n
  it('mocking the axios call to get posts should work', async () => {
    await flushPromises()
    expect(wrapper.vm.$data.people.length).toBe(1)
  })
})
