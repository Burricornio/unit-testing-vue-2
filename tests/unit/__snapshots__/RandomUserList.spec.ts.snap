// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomUserList suite Should match snapshots 1`] = `
<div>
  <div>
    <div class="person">
      <div class="person__header"><img src="" alt="[object Object]" class="rounded img-thumbnail">
        <div class="person__name">

        </div>
      </div>
      <div class="person__email"><a href="mailto:undefined"></a></div>
      <div class="person__address">
        <address>



        </address>
      </div>
    </div>
  </div>
</div>
`;
